crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -115,7 +119,42 @@ impl Trade {
                     },
                 }
             }
-            order::Kind::Liquidity => todo!(),
+            order::Kind::Liquidity => {
+                // Liquidity orders (including JIT) compute the executed amounts by linearly
+                // scaling the buy/sell amounts in the order.
+                match side {
+                    order::Side::Buy => Execution {
+                        buy: eth::Asset {
+                            amount: executed.into(),
+                            token: buy.token,
+                        },
+                        sell: eth::Asset {
+                            amount: executed
+                                .0
+                                .checked_mul(sell.amount)
```

Comment:
super nit: For some reason I had a super hard time parsing this. Could you consider switching these lines (and the equivalent lines in the other case)?
Then it would be `sell_amount * executed / buy_amount`. Which I would intuitively read like "scaling the `sell_amount` by some factor" where the factor is the relative fill amount of the order.
I might be the only one gets thrown off by this, though.